controller/replica_controller.go

```diff
@@ -417,6 +417,14 @@ func (rc *ReplicaController) GetBackingImagePathForReplicaStarting(r *longhorn.R
 		return "", nil
 	}
 	if _, exists := bi.Spec.Disks[r.Spec.DiskID]; !exists {
+		res, err := rc.ds.CanPutBackingImageOnDisk(bi, r.Spec.DiskID)
+		if err != nil {
+			log.Warnf("Failed to check if backing image %v can be put on disk %v", bi.Name, r.Spec.DiskID)
+		}
+		if !res {
+			log.Warnf("The backing image %v can not be put on the disk %v", bi.Name, r.Spec.DiskID)
+			return "", nil
+		}
```

Comment:
Yes you are right.
It should be an error since we already make it impossible to schedule replica to the node/disk where Backing Image cannot be stored